src/api/app/models/flag.rb

```diff
@@ -67,26 +93,13 @@ def to_s
     ret
   end
 
-  validates :flag, :presence => true
-  validates :position, :presence => true
-  validates_numericality_of :position, :only_integer => true
-
-  before_validation(:on => :create) do
-    if self.project
-      self.position = (self.project.flags.maximum(:position) || 0 ) + 1
-    elsif self.package
-      self.position = (self.package.flags.maximum(:position) || 0 ) + 1
-    end
+  def name
+    name = flag
+    name += "-#{repo}" unless repo.blank?
```

Comment:
it won't be a noticable performance gain, but it's a good habit to write performant code when having a choice ;)